### ISSUE TITLE
NotImplementedError: no replication rule for random_wrap

### ISSUE THREAD

**Issue by chenyue-max** [google/jax#22775]
### Description
```python
def shard_parallel(pr,args,out_tree):
    devices = jax.devices()
    mesh = Mesh(devices, ['x'])
    
    def sharded_eval_jaxpr(*args):
        def eval_jaxpr(*args):
            return jax.core.eval_jaxpr(pr.jaxpr, pr.consts, *args)
        
        return shard_map(eval_jaxpr, mesh, in_specs=P(None), out_specs=P(None))(*args)
    
    jitted_sharded_eval_jaxpr = jax.jit(sharded_eval_jaxpr)
    def evaluate_jaxpr_with_args(*args):
        return jitted_sharded_eval_jaxpr(*args)
    
    with ThreadPoolExecutor(max_workers=128) as executor:
        future = executor.submit(evaluate_jaxpr_with_args, *bargs)

    return future.result()
```
This code represents how I use shard_map and ran into a problem：
    
![Uploading image-20240731132809455.png…]()

### System info (python version, jaxlib version, accelerator, etc.)

jax:    0.4.6
jaxlib: 0.4.6
numpy:  1.23.5
python: 3.9.0 (default, Nov 15 2020, 14:28:56)  [GCC 7.3.0]
jax.devices (8 total, 8 local): [StreamExecutorGpuDevice(id=0, process_index=0, slice_index=0) StreamExecutorGpuDevice(id=1, process_index=0, slice_index=0) ... StreamExecutorGpuDevice(id=6, process_index=0, slice_index=0) StreamExecutorGpuDevice(id=7, process_index=0, slice_index=0)]
process_count: 1

**Comment by chenyue-max** [comment]
![image-20240731132809455](https://github.com/user-attachments/assets/792620c2-c25d-41b6-ab25-24c0bce6031c)

**Comment by jakevdp** [comment]
It looks like this has already been addressed by #20249, which is part of JAX v0.4.26 or newer.

It appears you are using JAX v0.4.6, so your best bet is probably to update to a more recent version.
